static/app/views/organizationIntegrations/sentryAppExternalForm.tsx

```diff
@@ -286,30 +286,37 @@ export class SentryAppExternalForm extends Component<Props, State> {
       if (isAsync) {
         fieldToPass.noOptionsMessage = () => 'Type to search';
       }
-    } else if (['text', 'textarea'].includes(fieldToPass.type || '')) {
+
+      if (field.depends_on) {
+        // check if this is dependent on other fields which haven't been set yet
+        const shouldDisable = field.depends_on.some(
+          dependentField => !hasValue(this.model.getValue(dependentField))
+        );
+        if (shouldDisable) {
+          fieldToPass = {...fieldToPass, disabled: true};
+        }
+      }
+    }
+    if (['text', 'textarea'].includes(fieldToPass.type || '')) {
       // Interpret the default if a getFieldDefault function is provided
-      let defaultValue = '';
+      let defaultValue = defaultResetValues.reduce((acc, curr) => {
```

Comment:
This block of code is duplicated on line 269. I also think that `reduce()` is a bed fit for what you're trying to do here. I'd suggest:
```tsx
getFieldValue = (field: FieldFromSchema) => {
    // Interpret the default if a getFieldDefault function is provided.
    const {resetValues} = this.props;
    return ((resetValues || {}).settings || []).find(value => value.name === field.name)?.value;
  };
...
let defaultValue = this.getFieldValue(field) || '';
```